src/backends/mysql/session.cpp

```diff
@@ -355,7 +357,11 @@ mysql_session_backend::mysql_session_backend(
     }
     if (reconnect_p)
     {
-        my_bool reconnect = 1;
+        #if MYSQL_VERSION_ID < 8
+            my_bool reconnect = 1;
+        #else
+            unsigned char reconnect = 1;
```

Comment:
Sorry, why `unsigned char` and not just `bool`? [The docs](https://dev.mysql.com/doc/c-api/8.0/en/mysql-options.html) seem to say that we should use `bool`.

Reply:
Hum yeah, you're right, it might be a problem with version 8.0.1:

>- since my_bool is gone in MySQL 8.0.0 and 8.0.1, it would be natural to
  "typedef bool my_bool", since bool is used everywhere my_bool was
  previously used, and the typedef allows to avoid massive application
  code changes;
>- now 8.0.2 re-introduces my_bool as an alias to char, which is
  compatible with the pre-8.0 definition of my_bool, but is incompatible
  with bool. Which conflicts with "typedef bool my_bool":
>drv_mysql.c:54:14: error: typedef redefinition with different types ('bool' vs 'char')
typedef bool my_bool;

I'll change it to bool